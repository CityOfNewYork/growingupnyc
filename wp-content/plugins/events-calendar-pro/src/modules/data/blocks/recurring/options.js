/**
 * External Dependencies
 */
import { __ } from '@wordpress/i18n';

/**
 * Internal Dependencies
 */
import * as constants from './constants';

//
// ─── RECURRENCE TYPES OPTIONS ───────────────────────────────────────────────────
//

export const RECURRENCE_TYPE_RULES_OPTIONS = [
	{
		label: constants.DAILY_LABEL,
		label_plural: constants.DAILY_LABEL_PLURAL,
		value: constants.DAILY
	},
	{
		label: constants.WEEKLY_LABEL,
		label_plural: constants.WEEKLY_LABEL_PLURAL,
		value : constants.WEEKLY
	},
	{
		label: constants.MONTHLY_LABEL,
		label_plural: constants.MONTHLY_LABEL_PLURAL,
		value: constants.MONTHLY
	},
	{
		label: constants.YEARLY_LABEL,
		label_plural: constants.YEARLY_LABEL_PLURAL,
		value: constants.YEARLY
	},
	{
		label: constants.SINGLE_LABEL,
		value: constants.SINGLE
	},
];

//
// ─── RECURRENCE FREQUENCY OPTIONS ───────────────────────────────────────────────
//

/**
 * Creates options for select element from 1 to max
 * @param {number} max The last number in the options list
 */
export const createNumericalOptions = ( max ) => (
	Array( max ).fill().map( ( _, index ) => ( {
		label: String( index + 1 ),
		value: index + 1,
	} ) )
);

export const DAILY_RECURRENCE_FREQUENCY_OPTIONS = createNumericalOptions( 6 );
export const WEEKLY_RECURRENCE_FREQUENCY_OPTIONS = createNumericalOptions( 6 );
export const MONTHLY_RECURRENCE_FREQUENCY_OPTIONS = createNumericalOptions( 12 );
export const YEARLY_RECURRENCE_FREQUENCY_OPTIONS = createNumericalOptions( 6 );

//
// ─── SERIES ENDS OPTIONS ────────────────────────────────────────────────────────
//

export const SERIES_ENDS_OPTIONS = [
	{ label: constants.ON_LABEL, value: constants.DATE },
	{ label: constants.AFTER_LABEL, value: constants.COUNT },
	{ label: constants.NEVER_LABEL, value: constants.NEVER },
];

//
// ─── DAYS OF THE WEEK OPTIONS ────────────────────────────────────────
//

export const DAYS_OF_THE_WEEK_OPTIONS = [
	{ label: constants.MONDAY_LABEL, value: constants.MONDAY },
	{ label: constants.TUESDAY_LABEL, value: constants.TUESDAY },
	{ label: constants.WEDNESDAY_LABEL, value: constants.WEDNESDAY },
	{ label: constants.THURSDAY_LABEL, value: constants.THURSDAY },
	{ label: constants.FRIDAY_LABEL, value: constants.FRIDAY },
	{ label: constants.SATURDAY_LABEL, value: constants.SATURDAY },
	{ label: constants.SUNDAY_LABEL, value: constants.SUNDAY },
	{ label: constants.DAY_LABEL, value: constants.DAY },
];

//
// ─── DAYS AND WEEKS OF THE MONTH OPTIONS ────────────────────────────────────────
//

export const DAYS_OF_THE_MONTH_OPTIONS = constants.DAYS_OF_THE_MONTH.map(
	( value ) => ( { label: String( value ), value } )
);

export const WEEKS_OF_THE_MONTH_OPTIONS = [
	{ label: constants.FIRST_LABEL, value: constants.FIRST },
	{ label: constants.SECOND_LABEL, value: constants.SECOND },
	{ label: constants.THIRD_LABEL, value: constants.THIRD },
	{ label: constants.FOURTH_LABEL, value: constants.FOURTH },
	{ label: constants.FIFTH_LABEL, value: constants.FIFTH },
	{ label: constants.LAST_LABEL, value: constants.LAST },
];

export const MONTH_DAYS_OPTIONS = [
	...WEEKS_OF_THE_MONTH_OPTIONS,
	...DAYS_OF_THE_MONTH_OPTIONS,
];

//
// ─── MONTHS OF THE YEAR OPTIONS ─────────────────────────────────────────────────
//

export const MONTHS_OF_THE_YEAR_OPTIONS = [
	{
		label: constants.JANUARY_LABEL,
		tag: constants.JANUARY_ABBR,
		value: constants.JANUARY,
	},
	{
		label: constants.FEBRUARY_LABEL,
		tag: constants.FEBRUARY_ABBR,
		value: constants.FEBRUARY,
	},
	{
		label: constants.MARCH_LABEL,
		tag: constants.MARCH_ABBR,
		value: constants.MARCH,
	},
	{
		label: constants.APRIL_LABEL,
		tag: constants.APRIL_ABBR,
		value: constants.APRIL,
	},
	{
		label: constants.MAY_LABEL,
		tag: constants.MAY_ABBR,
		value: constants.MAY,
	},
	{
		label: constants.JUNE_LABEL,
		tag: constants.JUNE_ABBR,
		value: constants.JUNE,
	},
	{
		label: constants.JULY_LABEL,
		tag: constants.JULY_ABBR,
		value: constants.JULY,
	},
	{
		label: constants.AUGUST_LABEL,
		tag: constants.AUGUST_ABBR,
		value: constants.AUGUST,
	},
	{
		label: constants.SEPTEMBER_LABEL,
		tag: constants.SEPTEMBER_ABBR,
		value: constants.SEPTEMBER,
	},
	{
		label: constants.OCTOBER_LABEL,
		tag: constants.OCTOBER_ABBR,
		value: constants.OCTOBER,
	},
	{
		label: constants.NOVEMBER_LABEL,
		tag: constants.NOVEMBER_ABBR,
		value: constants.NOVEMBER,
	},
	{
		label: constants.DECEMBER_LABEL,
		tag: constants.DECEMBER_ABBR,
		value: constants.DECEMBER,
	},
];

//
// ─── RECURRING MULTI DAY OPTIONS ────────────────────────────────────────────────
//

export const RECURRING_MULTI_DAY_OPTIONS = [
	{ label: constants.NEXT_DAY_LABEL, value: constants.NEXT_DAY },
	{ label: constants.SECOND_DAY_LABEL, value: constants.SECOND_DAY },
	{ label: constants.THIRD_DAY_LABEL, value: constants.THIRD_DAY },
	{ label: constants.FOURTH_DAY_LABEL, value: constants.FOURTH_DAY },
	{ label: constants.FIFTH_DAY_LABEL, value: constants.FIFTH_DAY },
	{ label: constants.SIXTH_DAY_LABEL, value: constants.SIXTH_DAY },
	{ label: constants.SEVENTH_DAY_LABEL, value: constants.SEVENTH_DAY },
];
