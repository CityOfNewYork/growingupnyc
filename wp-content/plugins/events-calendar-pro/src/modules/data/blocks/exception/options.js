/**
 * External Dependencies
 */
import { __ } from '@wordpress/i18n';

/**
 * Internal Dependencies
 */
import * as constants from '@moderntribe/events-pro/data/blocks/recurring/constants';

export const EXCEPTION_OCCURRENCE_OPTIONS = [
	{ label: __( 'Daily', 'events-gutenberg' ), value: constants.DAILY },
	{ label: __( 'Weekly', 'events-gutenberg' ), value: constants.WEEKLY },
	{ label: __( 'Monthly', 'events-gutenberg' ), value: constants.MONTHLY },
	{ label: __( 'Yearly', 'events-gutenberg' ), value: constants.YEARLY },
	{ label: __( 'Single Exception', 'events-gutenberg' ), value: constants.SINGLE },
];
