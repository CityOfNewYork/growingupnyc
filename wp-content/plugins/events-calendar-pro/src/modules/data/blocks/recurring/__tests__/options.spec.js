/**
 * Internal Dependencies
 */
import * as options from '../options';

describe( 'Recurrence options', () => {
	describe( 'Recurrence type options', () => {
		test( 'recurrence type rules options', () => {
			expect( options.RECURRENCE_TYPE_RULES_OPTIONS ).toMatchSnapshot();
		} );
	} );

	describe( 'Recurrence frequency options', () => {
		test( 'createNumericalOptions', () => {
			expect( options.createNumericalOptions( 5 ) ).toMatchSnapshot();
		} );

		test( 'daily recurrence frequency options', () => {
			expect( options.DAILY_RECURRENCE_FREQUENCY_OPTIONS ).toMatchSnapshot();
		} );

		test( 'weekly recurrence frequency options', () => {
			expect( options.WEEKLY_RECURRENCE_FREQUENCY_OPTIONS ).toMatchSnapshot();
		} );

		test( 'monthly recurrence frequency options', () => {
			expect( options.MONTHLY_RECURRENCE_FREQUENCY_OPTIONS ).toMatchSnapshot();
		} );

		test( 'yearly recurrence frequency options', () => {
			expect( options.YEARLY_RECURRENCE_FREQUENCY_OPTIONS ).toMatchSnapshot();
		} );
	} );

	describe( 'Series ends options', () => {
		test( 'series ends options', () => {
			expect( options.SERIES_ENDS_OPTIONS ).toMatchSnapshot();
		} );
	} );

	describe( 'Days of the week options', () => {
		test( 'days of the week options', () => {
			expect( options.DAYS_OF_THE_WEEK_OPTIONS ).toMatchSnapshot();
		} );
	} );

	describe( 'Days of the month options', () => {
		test( 'days of the month options', () => {
			expect( options.DAYS_OF_THE_MONTH_OPTIONS ).toMatchSnapshot();
		} );

		test( 'weeks of the month options', () => {
			expect( options.WEEKS_OF_THE_MONTH_OPTIONS ).toMatchSnapshot();
		} );

		test( 'month days options', () => {
			expect( options.MONTH_DAYS_OPTIONS ).toMatchSnapshot();
		} );
	} );

	describe( 'Months of the year options', () => {
		test( 'months of the year options', () => {
			expect( options.MONTHS_OF_THE_YEAR_OPTIONS ).toMatchSnapshot();
		} );
	} );

	describe( 'Recurring multi day options', () => {
		test( 'recurring multi day options', () => {
			expect( options.RECURRING_MULTI_DAY_OPTIONS ).toMatchSnapshot();
		} );
	} );
} );
