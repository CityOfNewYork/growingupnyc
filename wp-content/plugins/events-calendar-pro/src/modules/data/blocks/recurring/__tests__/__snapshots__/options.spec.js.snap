// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recurrence options Days of the month options days of the month options 1`] = `
Array [
  Object {
    "label": "1",
    "value": 1,
  },
  Object {
    "label": "2",
    "value": 2,
  },
  Object {
    "label": "3",
    "value": 3,
  },
  Object {
    "label": "4",
    "value": 4,
  },
  Object {
    "label": "5",
    "value": 5,
  },
  Object {
    "label": "6",
    "value": 6,
  },
  Object {
    "label": "7",
    "value": 7,
  },
  Object {
    "label": "8",
    "value": 8,
  },
  Object {
    "label": "9",
    "value": 9,
  },
  Object {
    "label": "10",
    "value": 10,
  },
  Object {
    "label": "11",
    "value": 11,
  },
  Object {
    "label": "12",
    "value": 12,
  },
  Object {
    "label": "13",
    "value": 13,
  },
  Object {
    "label": "14",
    "value": 14,
  },
  Object {
    "label": "15",
    "value": 15,
  },
  Object {
    "label": "16",
    "value": 16,
  },
  Object {
    "label": "17",
    "value": 17,
  },
  Object {
    "label": "18",
    "value": 18,
  },
  Object {
    "label": "19",
    "value": 19,
  },
  Object {
    "label": "20",
    "value": 20,
  },
  Object {
    "label": "21",
    "value": 21,
  },
  Object {
    "label": "22",
    "value": 22,
  },
  Object {
    "label": "23",
    "value": 23,
  },
  Object {
    "label": "24",
    "value": 24,
  },
  Object {
    "label": "25",
    "value": 25,
  },
  Object {
    "label": "26",
    "value": 26,
  },
  Object {
    "label": "27",
    "value": 27,
  },
  Object {
    "label": "28",
    "value": 28,
  },
  Object {
    "label": "29",
    "value": 29,
  },
  Object {
    "label": "30",
    "value": 30,
  },
  Object {
    "label": "31",
    "value": 31,
  },
]
`;

exports[`Recurrence options Days of the month options month days options 1`] = `
Array [
  Object {
    "label": "First",
    "value": "first",
  },
  Object {
    "label": "Second",
    "value": "second",
  },
  Object {
    "label": "Third",
    "value": "third",
  },
  Object {
    "label": "Fourth",
    "value": "fourth",
  },
  Object {
    "label": "Fifth",
    "value": "fifth",
  },
  Object {
    "label": "Last",
    "value": "last",
  },
  Object {
    "label": "1",
    "value": 1,
  },
  Object {
    "label": "2",
    "value": 2,
  },
  Object {
    "label": "3",
    "value": 3,
  },
  Object {
    "label": "4",
    "value": 4,
  },
  Object {
    "label": "5",
    "value": 5,
  },
  Object {
    "label": "6",
    "value": 6,
  },
  Object {
    "label": "7",
    "value": 7,
  },
  Object {
    "label": "8",
    "value": 8,
  },
  Object {
    "label": "9",
    "value": 9,
  },
  Object {
    "label": "10",
    "value": 10,
  },
  Object {
    "label": "11",
    "value": 11,
  },
  Object {
    "label": "12",
    "value": 12,
  },
  Object {
    "label": "13",
    "value": 13,
  },
  Object {
    "label": "14",
    "value": 14,
  },
  Object {
    "label": "15",
    "value": 15,
  },
  Object {
    "label": "16",
    "value": 16,
  },
  Object {
    "label": "17",
    "value": 17,
  },
  Object {
    "label": "18",
    "value": 18,
  },
  Object {
    "label": "19",
    "value": 19,
  },
  Object {
    "label": "20",
    "value": 20,
  },
  Object {
    "label": "21",
    "value": 21,
  },
  Object {
    "label": "22",
    "value": 22,
  },
  Object {
    "label": "23",
    "value": 23,
  },
  Object {
    "label": "24",
    "value": 24,
  },
  Object {
    "label": "25",
    "value": 25,
  },
  Object {
    "label": "26",
    "value": 26,
  },
  Object {
    "label": "27",
    "value": 27,
  },
  Object {
    "label": "28",
    "value": 28,
  },
  Object {
    "label": "29",
    "value": 29,
  },
  Object {
    "label": "30",
    "value": 30,
  },
  Object {
    "label": "31",
    "value": 31,
  },
]
`;

exports[`Recurrence options Days of the month options weeks of the month options 1`] = `
Array [
  Object {
    "label": "First",
    "value": "first",
  },
  Object {
    "label": "Second",
    "value": "second",
  },
  Object {
    "label": "Third",
    "value": "third",
  },
  Object {
    "label": "Fourth",
    "value": "fourth",
  },
  Object {
    "label": "Fifth",
    "value": "fifth",
  },
  Object {
    "label": "Last",
    "value": "last",
  },
]
`;

exports[`Recurrence options Days of the week options days of the week options 1`] = `
Array [
  Object {
    "label": "Monday",
    "value": "monday",
  },
  Object {
    "label": "Tuesday",
    "value": "tuesday",
  },
  Object {
    "label": "Wednesday",
    "value": "wednesday",
  },
  Object {
    "label": "Thursday",
    "value": "thursday",
  },
  Object {
    "label": "Friday",
    "value": "friday",
  },
  Object {
    "label": "Saturday",
    "value": "saturday",
  },
  Object {
    "label": "Sunday",
    "value": "sunday",
  },
  Object {
    "label": "Day",
    "value": "day",
  },
]
`;

exports[`Recurrence options Months of the year options months of the year options 1`] = `
Array [
  Object {
    "label": "January",
    "tag": "Jan",
    "value": "january",
  },
  Object {
    "label": "February",
    "tag": "Feb",
    "value": "february",
  },
  Object {
    "label": "March",
    "tag": "Mar",
    "value": "march",
  },
  Object {
    "label": "April",
    "tag": "Apr",
    "value": "april",
  },
  Object {
    "label": "May",
    "tag": "May",
    "value": "may",
  },
  Object {
    "label": "June",
    "tag": "Jun",
    "value": "june",
  },
  Object {
    "label": "July",
    "tag": "Jul",
    "value": "july",
  },
  Object {
    "label": "August",
    "tag": "Aug",
    "value": "august",
  },
  Object {
    "label": "September",
    "tag": "Sep",
    "value": "september",
  },
  Object {
    "label": "October",
    "tag": "Oct",
    "value": "october",
  },
  Object {
    "label": "November",
    "tag": "Nov",
    "value": "november",
  },
  Object {
    "label": "December",
    "tag": "Dec",
    "value": "december",
  },
]
`;

exports[`Recurrence options Recurrence frequency options createNumericalOptions 1`] = `
Array [
  Object {
    "label": "1",
    "value": 1,
  },
  Object {
    "label": "2",
    "value": 2,
  },
  Object {
    "label": "3",
    "value": 3,
  },
  Object {
    "label": "4",
    "value": 4,
  },
  Object {
    "label": "5",
    "value": 5,
  },
]
`;

exports[`Recurrence options Recurrence frequency options daily recurrence frequency options 1`] = `
Array [
  Object {
    "label": "1",
    "value": 1,
  },
  Object {
    "label": "2",
    "value": 2,
  },
  Object {
    "label": "3",
    "value": 3,
  },
  Object {
    "label": "4",
    "value": 4,
  },
  Object {
    "label": "5",
    "value": 5,
  },
  Object {
    "label": "6",
    "value": 6,
  },
]
`;

exports[`Recurrence options Recurrence frequency options monthly recurrence frequency options 1`] = `
Array [
  Object {
    "label": "1",
    "value": 1,
  },
  Object {
    "label": "2",
    "value": 2,
  },
  Object {
    "label": "3",
    "value": 3,
  },
  Object {
    "label": "4",
    "value": 4,
  },
  Object {
    "label": "5",
    "value": 5,
  },
  Object {
    "label": "6",
    "value": 6,
  },
  Object {
    "label": "7",
    "value": 7,
  },
  Object {
    "label": "8",
    "value": 8,
  },
  Object {
    "label": "9",
    "value": 9,
  },
  Object {
    "label": "10",
    "value": 10,
  },
  Object {
    "label": "11",
    "value": 11,
  },
  Object {
    "label": "12",
    "value": 12,
  },
]
`;

exports[`Recurrence options Recurrence frequency options weekly recurrence frequency options 1`] = `
Array [
  Object {
    "label": "1",
    "value": 1,
  },
  Object {
    "label": "2",
    "value": 2,
  },
  Object {
    "label": "3",
    "value": 3,
  },
  Object {
    "label": "4",
    "value": 4,
  },
  Object {
    "label": "5",
    "value": 5,
  },
  Object {
    "label": "6",
    "value": 6,
  },
]
`;

exports[`Recurrence options Recurrence frequency options yearly recurrence frequency options 1`] = `
Array [
  Object {
    "label": "1",
    "value": 1,
  },
  Object {
    "label": "2",
    "value": 2,
  },
  Object {
    "label": "3",
    "value": 3,
  },
  Object {
    "label": "4",
    "value": 4,
  },
  Object {
    "label": "5",
    "value": 5,
  },
  Object {
    "label": "6",
    "value": 6,
  },
]
`;

exports[`Recurrence options Recurrence type options recurrence type rules options 1`] = `
Array [
  Object {
    "label": "Day",
    "value": "daily",
  },
  Object {
    "label": "Week",
    "value": "weekly",
  },
  Object {
    "label": "Month",
    "value": "monthly",
  },
  Object {
    "label": "Year",
    "value": "yearly",
  },
  Object {
    "label": "Single Recurrence",
    "value": "single",
  },
]
`;

exports[`Recurrence options Recurring multi day options recurring multi day options 1`] = `
Array [
  Object {
    "label": "Next day",
    "value": "next_day",
  },
  Object {
    "label": "2nd day",
    "value": "second_day",
  },
  Object {
    "label": "3rd day",
    "value": "third_day",
  },
  Object {
    "label": "4th day",
    "value": "fourth_day",
  },
  Object {
    "label": "5th day",
    "value": "fifth_day",
  },
  Object {
    "label": "6th day",
    "value": "sixth_day",
  },
  Object {
    "label": "7th day",
    "value": "seventh_day",
  },
]
`;

exports[`Recurrence options Series ends options series ends options 1`] = `
Array [
  Object {
    "label": "On",
    "value": "date",
  },
  Object {
    "label": "After",
    "value": "count",
  },
  Object {
    "label": "Never",
    "value": "never",
  },
]
`;
